central/complianceoperator/v2/checkresults/service/service_impl.go

```diff
@@ -385,34 +345,31 @@ func (s *serviceImpl) GetComplianceProfileClusterResults(ctx context.Context, re
 	}, nil
 }
 
-func (s *serviceImpl) mapScanConfigToID(ctx context.Context, scanResults []*storage.ComplianceOperatorCheckResultV2) (map[string]string, error) {
-	scanConfigToIDs := make(map[string]string, len(scanResults))
-	for _, result := range scanResults {
-		if _, found := scanConfigToIDs[result.ScanConfigName]; !found {
-			config, err := s.scanConfigDS.GetScanConfigurationByName(ctx, result.ScanConfigName)
-			if err != nil {
-				return nil, errors.Errorf("Unable to retrieve valid compliance scan configuration %q", result.ScanConfigName)
-			}
-			scanConfigToIDs[result.ScanConfigName] = config.GetId()
-		}
-	}
-
-	return scanConfigToIDs, nil
-}
-
-func (s *serviceImpl) searchComplianceCheckResults(ctx context.Context, parsedQuery *v1.Query) (*v2.ListComplianceScanResultsResponse, error) {
+func (s *serviceImpl) searchComplianceCheckResults(ctx context.Context, parsedQuery *v1.Query, countQuery *v1.Query) (*v2.ListComplianceResultsResponse, error) {
 	scanResults, err := s.complianceResultsDS.SearchComplianceCheckResults(ctx, parsedQuery)
 	if err != nil {
 		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to retrieve compliance scan results for query %v", parsedQuery)
 	}
 
-	// Need to look up the scan config IDs to return with the results.
-	scanConfigToIDs, err := s.mapScanConfigToID(ctx, scanResults)
+	checkToRule := make(map[string]string, len(scanResults))
+	for _, result := range scanResults {
+		rules, err := s.ruleDS.SearchRules(ctx, search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorRuleRef, result.GetRuleRefId()).ProtoQuery())
```

Comment:
Are you using the `RuleRefId` here to get the rule by cluster?
Or is this just the only link from a result to a rule?

Reply:
It is the only single field link from a result to a rule.  Thus the simplest.